api-bucket-lifecycle.go

```diff
@@ -48,15 +49,16 @@ func (c *Client) SetBucketLifecycle(ctx context.Context, bucketName string, conf
 	}
 
 	// Save the updated lifecycle.
-	return c.putBucketLifecycle(ctx, bucketName, buf)
+	return c.putBucketLifecycle(ctx, bucketName, buf, expiryRuleRemoved)
 }
 
 // Saves a new bucket lifecycle.
-func (c *Client) putBucketLifecycle(ctx context.Context, bucketName string, buf []byte) error {
+func (c *Client) putBucketLifecycle(ctx context.Context, bucketName string, buf []byte, expiryRuleRemoved bool) error {
 	// Get resources properly escaped and lined up before
 	// using them in http request.
 	urlValues := make(url.Values)
 	urlValues.Set("lifecycle", "")
+	urlValues.Set("expiryRuleRemoved", strconv.FormatBool(expiryRuleRemoved))
```

Comment:
also why does it say `expiryRuleRemoved` ?

Reply:
Actually `SetBucketLifecycle` gets invoked for ILM add/edit as well as ILM remove scenarios. So to differentiate setting of ILM rule whether its removal or not, I set the flag. It gets passed as true only in case of ILM delete scenario. Could there be a better name, we can change this name.